synapse/federation/federation_client.py

```diff
@@ -371,6 +393,13 @@ async def get_pdu(
             now = self._clock.time_msec()
```

Comment:
wait... does this loop *always* try all the destinations, even if the first one works? that would be a substantial bug, if it was ever called with more than one destination, which I don't think it is...

Reply:
Seems like it is bugged. I can add a `break` or refactor to do the `return` inside the `for-loop` in a separate PR after this merges ⏩

Reply:
I've added a `FIXME` comment there for now so we don't forget or at least will know to tackle it at some point. I plan to tackle in a fast-follow-up after this PR since it's a separate bug. Feel free to push to include it here.